### PR TITLE
feat(atomic): extract result list interface

### PR DESCRIPTION
**Pull request by liviarett**
https://coveord.atlassian.net/browse/KIT-1647

I reused the jira ticket, hope it works 🙏 
@olamothe let me know what you think. I had to make a bunch of stuff public to be able to extract it to the interface, but can change that!

### CODE REVIEW

**Review by btaillon — APPROVED** [review]

File: packages/atomic/src/components/result-lists/result-list-common.tsx [hunk 1]
```diff
@@ -29,6 +29,26 @@ import {ListDisplayResultsPlaceholder} from './list-display-results-placeholder'
 import {once} from '../../utils/utils';
 import {updateBreakpoints} from '../../utils/replace-breakpoint';
 import {isAppLoaded, setLoadingFlag} from '../../utils/store';
+
+export interface BaseResultList<T = FoldedResult | Result> {
```
Comment (ThibodeauJF) [inline packages/atomic/src/components/result-lists/result-list-common.tsx:33]:
```suggestion
export interface BaseResultList<T = FoldedResult | Result> extends InitializableComponent {
```

**Review by ThibodeauJF — APPROVED** [review]
💯

File: packages/atomic/src/components/result-lists/result-list-common.tsx (hunk 1 above)
Comment (liviarett) [reply packages/atomic/src/components/result-lists/result-list-common.tsx:33]: nice!! will update in the code

**Review by olamothe — APPROVED** [review]
Neat ! That way we'll have the TypeScript compiler correctly complain to use when we forget an option on one of the "result list"

**Comment by liviarett** [comment]
I made a little booboo with the rebasing.. will fix tomorrow!!

**Pull request merged**